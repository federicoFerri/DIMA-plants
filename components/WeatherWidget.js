import React from 'react';
import { StyleSheet, Text, View, Image, SafeAreaView} from 'react-native';

//widget that shows the weather and the temperature of a location
//Required props
//@icon : url of the image of the weather condition
//@condition : string that explains the weather condition
//@temperature : number of the temperature in celsius
class WeatherWidget extends React.Component {
    render() {
        return (
            <SafeAreaView style={{flexDirection:'row', justifyContent: 'center', alignItems: 'center'}}>
                <Image 
                    style={{resizeMode: 'stretch', width: 45, height: 44}}
                    source={{uri: 'https:'+ this.props.icon}}
                />
                <View>
                <Text style={{fontSize: 11, color: '#000', fontFamily:'Comfortaa'}}>{this.props.condition}</Text>
                <Text style={{fontSize: 11, color: '#000', fontFamily:'Comfortaa'}}>{this.props.temperature}Â°C</Text>
                <Text style={{fontSize: 11, color: '#000', fontFamily:'Comfortaa'}}>from weather api</Text>
                </View> 
                
            </SafeAreaView>
        )
    }
}

export default WeatherWidget;