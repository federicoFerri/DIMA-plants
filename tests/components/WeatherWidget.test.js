import React from 'react';
import renderer from 'react-test-renderer';

import WeatherWidget from '../../components/WeatherWidget'

describe('<WeatherWidget />', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<WeatherWidget />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});