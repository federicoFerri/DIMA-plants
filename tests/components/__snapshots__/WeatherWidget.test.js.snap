// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherWidget /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https:undefined",
      }
    }
    style={
      Object {
        "height": 44,
        "resizeMode": "stretch",
        "width": 45,
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Comfortaa",
          "fontSize": 11,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Comfortaa",
          "fontSize": 11,
        }
      }
    >
      Â°C
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Comfortaa",
          "fontSize": 11,
        }
      }
    >
      from weather api
    </Text>
  </View>
</RCTSafeAreaView>
`;
